Uploading NER notebook

anjleeg-gcai opened the pull request.

payoto commented on notebooks/name-entity-extraction.ipynb:
```diff
@@ -0,0 +1,785 @@
+{
```
<p>Move this before "Lets load"</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/optimum-graphcore-fork/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='9'/>

payoto replied:
<p>"the following function can build our app:"</p><p> </p><p>replace with:</p><p> </p><p>"The <code>app_for_pipeline</code> function will build a small app which includes a text prompt and will render the entities which were identified."</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/optimum-graphcore-fork/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='15'/>

payoto replied:
<p>Too many exclamation marks. Use very sparingly.</p><p> </p><p>Remove "major" (in major languages)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/optimum-graphcore-fork/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='20'/>

payoto replied:
<p>Add a concluding phrase about the performance of the previous model, "it works great on the different languages"</p><p> </p><p> </p><p>Move this to the next section.</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/optimum-graphcore-fork/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='26'/>

payoto replied:
<p>Same comment as before (move  down, conclude on current model)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/optimum-graphcore-fork/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='33'/>

payoto replied:
<p>performs not preforms</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/optimum-graphcore-fork/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='34'/>

payoto replied:
<p>injuries, not inguries</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/optimum-graphcore-fork/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='36'/>

payoto commented on notebooks/name-entity-extraction.ipynb:
```diff
@@ -0,0 +1,785 @@
+{
```
<pre class="ql-syntax" spellcheck="false">Line #1.    multilingual_examples = ["A Budapest, il y a une grande piscine que les touristes visitent.",
</pre><p> </p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/optimum-graphcore-fork/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='23'/>

anjleeg-gcai replied:
<p>Ty, I forgot to actually put that in translate instead of using my brain to practice for fun.</p>

payoto commented on notebooks/name-entity-extraction.ipynb:
```diff
@@ -0,0 +1,785 @@
+{
```
<p>how how</p><p> </p><p> super fast</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/optimum-graphcore-fork/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='40'/>

anjleeg-gcai replied:
<p>Not sure if I understand your comment  </p>

payoto commented:
Looks really good.

Some minor comments, and you're missing something about describing the responsiveness of the UI being enabled by the inference speed of the IPU. This makes the IPU fast enough to be used by APIs that need to be used in real time by users.

anjleeg-gcai commented:
Hi, I've made the changes you suggested @payoto , including rewriting the conclusion with what you mentioned in your above comment, hopefully this is okay now? Will send to Jamie for review too.

payoto commented:
I think the responsiveness element should be highlighted earlier in the notebook, after the first GUI.

In terms of other things:

- detaching the pipelines from the IPUs (I've not checked if you've done it)

JamesRandom commented on notebooks/name-entity-extraction.ipynb:
```diff
@@ -0,0 +1,422 @@
+    "\n",
+    "Integration of the Graphcore Intelligence Processing Unit (IPU) and the Hugging Face transformer library means that it only takes a few lines of code to perform complex tasks which require deep learning.\n",
+    "\n",
+    "In this notebook we perform **name entity extraction (NER)**  also known as token classification: we use natural language processing models to classify the words inside the prompt. \n",
```
I would prefer "a prompt" (as we haven't previously talked about "the" prompt

JamesRandom commented on notebooks/name-entity-extraction.ipynb:
```diff
@@ -0,0 +1,422 @@
+    "\n",
+    "\n",
+    "The ease of use of the `pipeline` interface lets us quickly experiment with the pre-trained models and identify which one will work best.\n",
+    "This simple interface means that it is extremely easy to access the fast inference performance of the IPU on your application.\n",
```
We advise avoiding things like "[extremely] easy" because if the user has problems (whether our fault or their fault) they will feel even more frustrated if they have been told it will be easy. Maybe something more neutral like "straightforward"

JamesRandom commented on notebooks/name-entity-extraction.ipynb:
```diff
@@ -0,0 +1,422 @@
+   "metadata": {},
+   "source": [
+    "The simplest way to use a model on the IPU is to use the `pipeline` function from the `transformers` library.\n",
+    "It provides a set of models which have been validated to work on a given task. \n",
```
I found "It provides" a bit confusing — it looks like "It" is referring to "the pipeline function". Is that correct? Or does "It" mean the transformers library?

anjleeg-gcai commented:
Hi I've made the suggested changes, and I have also tested against sdk 3.1 so this should be ready to go.

payoto commented on notebooks/name-entity-extraction.ipynb:
```diff
@@ -0,0 +1,452 @@
+{
```
<p>Can you update that screenshot to one that corresponds to the final app of this notebook? (with the medical stuff)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/optimum-graphcore-fork/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='0'/>

payoto replied:
<p>I don't think you committed the image?</p>

payoto commented on notebooks/name-entity-extraction.ipynb:
```diff
@@ -0,0 +1,452 @@
+{
```
<blockquote>Using them, we will experiment with other models and applications which are just as responsive as the first, but more directed for specific purposes.</blockquote><p>A bit too general, Can you say which specific purposes?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/optimum-graphcore-fork/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='21'/>

payoto replied:
<p>spell: feild -&gt; field</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/optimum-graphcore-fork/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='28'/>

payoto replied:
<p>In the conclusion:</p><p>NIT: "showed us how how" -&gt; "showed how easy..."</p><blockquote>showing us how easy and powerful it is when being used as a back end for an API.</blockquote><p>NIT: "it is when being used" -&gt; "when used"</p><p> </p><p>Sentence 2: Be more specific about which purposes and tasks, and add a callout to other tasks (with a link to the HUB)</p><p> </p><p>Sentence 3: would also benefit from being more specific, maybe mention real-time interaction for a person? Say in what way it is powerful (speed) and easy (Numbers of lines of code)? Maybe mention that this approach would also support other applications, like translation? sentiment analysis?</p><p> </p><p>Conclusion needs a next step/opening: go and see another related notebook? Maybe the one on finetuning for token classification?</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/graphcore/optimum-graphcore-fork/pull/17/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='38'/>

payoto reviewed (changes requested):
Sorry to ask for more changes, but I expect it to be the last round!

JamesRandom commented on notebooks/name-entity-extraction.ipynb:
```diff
@@ -0,0 +1,452 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Real time Name entity recognition on the IPU\n",
```
Title case: Real Time Name Entity Recognition on the IPU

JamesRandom commented on notebooks/name-entity-extraction.ipynb:
```diff
@@ -0,0 +1,452 @@
+    "\n",
+    "Using gradio we are able to tell which words are being correctly extracted and identified for different categories using our model.\n",
+    "\n",
+    "This is all aided by the IPU which is quickly processing our inputs and returning the models outputs to create a very responsive interface.\n",
```
Either "model's outputs" or "model outputs"

JamesRandom commented on notebooks/name-entity-extraction.ipynb:
```diff
@@ -0,0 +1,452 @@
+    "\n",
+    "This is all aided by the IPU which is quickly processing our inputs and returning the models outputs to create a very responsive interface.\n",
+    "\n",
+    "Next we must detach our model from the IPU to release resources, you can learn more about resource management for the IPU [here](https://github.com/huggingface/optimum-graphcore/blob/main/notebooks/managing_ipu_resources.ipynb)."
```
Avoid "here" links; they don't give users any idea what the link is and are bad for accessibility.
```
in our [managing resources notebook](https://github.com/huggingface/optimum-graphcore/blob/main/notebooks/managing_ipu_resources.ipynb).
```

JamesRandom commented on notebooks/name-entity-extraction.ipynb:
```diff
@@ -0,0 +1,452 @@
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "As we can see the results of the model are very descriptive and we are able to test out different inputs in our pipeline to enable us to identify important patient information using the biomedical model.\n",
```
Comma after "As we can see"

payoto commented:
I added this notebook to the Paperspace test suite and it hit the following error: `NameError: name 'ner_pipeline_medical' is not defined` it seems to be a typo according to @rahult-graphcore

anjleeg-gcai commented:
Hi, thanks for all the suggestions @payoto  and @JamesRandom.

I've made changes and rewritten some parts, hopefully this is okay now, struggled a bit with the conclusion and happy to edit further if it needs it. 

Also fixed the typo ^

payoto reviewed (approved):
Looks good! I think we can call it there
